### PR TITLE
Added hardfork to RpcClient

### PR DESCRIPTION
**Pull request by cschuchardt88**
Added hardfork to RpcClient and tested

This is for #823 implementation of hardfork settings in `getversion` rpc method.

### CODE REVIEW

File: tests/Neo.Network.RPC.Tests/RpcTestCases.json [hunk 1]
```diff
@@ -2491,7 +2491,13 @@
           "addressversion": 0,
           "maxtransactionsperblock": 0,
           "memorypoolmaxtransactions": 0,
-          "initialgasdistribution": 0
+          "initialgasdistribution": 0,
+          "hardforks": [
+            {
+              "name": "HF_Aspidochelone",
```
Comment (roman-khimov) [inline tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: BTW, speaking of #822, can we avoid `HF_` prefixes here? It's `hardforks` section already.

Comment (cschuchardt88) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: but it based off Enum `Hardfork`, So its parses the name, i guess i could put value, i believe that works.

Comment (cschuchardt88) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: oh i see what your saying, Thats how its used in #823.

```
public enum Hardfork : byte
{
    HF_Aspidochelone,
    HF_Basilisk
}
```

Comment (roman-khimov) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: Yeah, it's not a client problem and not something client should be bothered with. It's just that I've only noticed it here. But server-side it'd probably be a bit better.

Comment (cschuchardt88) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: https://github.com/neo-project/neo-modules/blob/63e3ec3dc4092ea83f010b1fe9704a70129fbcda/src/RpcServer/RpcServer.Node.cs#L88

Code uses key of the enum which in our case is `HF_Aspidochelone` and `HF_Basilisk`. We would have to change the server to use the value instead. If you would like me to proceed in changing the server let me know.

Comment (shargon) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: > Yeah, it's not a client problem and not something client should be bothered with. It's just that I've only noticed it here. But server-side it'd probably be a bit better.

Why we need it in server? We can remove it from both

Comment (cschuchardt88) [reply tests/Neo.Network.RPC.Tests/RpcTestCases.json:2497]: remember this #823 someone merged it

**Comment by Liaojinghui** [comment]
Good to go? ifs its already in the server.

**Review by shargon — APPROVED** [review]

**Pull request merged**